connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaConfigBackingStore.java

```diff
@@ -622,19 +620,20 @@ private ConnectorTaskId parseTaskId(String key) {
     }
 
     /**
-     * Given task configurations, get a set of integer task IDs organized by connector name.
+     * Given task configurations, get a set of integer task IDs for the connector.
      */
-    private Map<String, Set<Integer>> taskIdsByConnector(Map<ConnectorTaskId, Map<String, String>> configs) {
-        Map<String, Set<Integer>> connectorTaskIds = new HashMap<>();
-        if (configs == null)
-            return connectorTaskIds;
+    private Set<Integer> taskIds(String connector, Map<ConnectorTaskId, Map<String, String>> configs) {
+        Set<Integer> tasks = new TreeSet<>();
+        if (configs == null) {
+            return tasks;
+        }
         for (Map.Entry<ConnectorTaskId, Map<String, String>> taskConfigEntry : configs.entrySet()) {
             ConnectorTaskId taskId = taskConfigEntry.getKey();
-            if (!connectorTaskIds.containsKey(taskId.connector()))
-                connectorTaskIds.put(taskId.connector(), new TreeSet<Integer>());
-            connectorTaskIds.get(taskId.connector()).add(taskId.task());
+            if (taskId.connector().equals(connector)) {
```

Comment:
I guess the awkwardness here is caused by the fact that the caller _could_ submit a config set which includes tasks for a different connector. Instead of silently ignoring this case when it happens, I think it might be better to either to 1) assume that the we won't do this (which is probably fine since putTaskConfigs is an internal API), or 2) validate the argument in `putTaskConfigs` and raise an exception.

Alternatively, we could change the signature of `putTaskConfigs` to the following:
```java
public void putTaskConfigs(String connector, Map<Integer, Map<String, String>> configs);
```
The advantage is that it saves the need to create this new set. What do you think?